include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -4975,19 +4988,33 @@ MGTwoLevelTransferNonNested<dim, VectorType>::prolongate_and_add_internal_comp(
             {
               result = {};
               for (unsigned int k = 0; k < n_entries; ++k)
-                result += evaluation_point_results[ptr[j] + k];
+                {
+                  const unsigned int index = recv_permutation_inv[ptr[j] + k];
+
+                  for (unsigned int c = 0; c < n_components; ++c)
+                    result[c] +=
+                      evaluation_point_results[index * n_components + c];
+                }
               if (n_entries <= typical_weights.size())
-                result *= typical_weights[n_entries - 1];
+                for (unsigned int c = 0; c < n_components; ++c)
+                  result[c] *= typical_weights[n_entries - 1];
               else
-                result /= Number(n_entries);
+                for (unsigned int c = 0; c < n_components; ++c)
+                  result[c] /= Number(n_entries);
```

Comment:
Could you please write this as
```suggestion
                  result[c] *= Number(1) / Number(n_entries);
```